### DISCUSSION TITLE
Question about imports

### DISCUSSION THREAD

**Discussion by mike-ward** [vlang/v#21727] [Questions and Answers]
Importing `encoding.base32` yields this warning
```
/Users/mike/Documents/github/v/vlib/encoding/base32/base32.v:289:13: notice: an implicit clone of the slice was done here
  287 |             }
  288 |             in0 := src[0]
  289 |             src = src[1..]
      |                      ~~~~~
  290 |             if in0 == enc.padding_char && j >= 2 && src.len < 8 {
  291 |                 // We`ve reached the end and there`s padding
Details: /Users/mike/Documents/github/v/vlib/encoding/base32/base32.v:289:13: details: To silence this notice, use either an explicit `a[..].clone()`,
or use an explicit `unsafe{ a[..] }`, if you do not want a copy of the slice.
  287 |             }
  288 |             in0 := src[0]
  289 |             src = src[1..]
      |                      ~~~~~
  290 |             if in0 == enc.padding_char && j >= 2 && src.len < 8 {
  291 |                 // We`ve reached the end and there`s padding
```
This implies `vlib` modules are compiled from source on import. This is different from what I'm accustomed to in `C#`. 

What if I only had the V compiler and not the source repository? Would I see the same error?

**Comment by mike-ward** [comment]
Guess I'm going to answer my own question here. I downloaded the prebuilt mac version and sure enough, the source for `vlib` is included in the package. Makes sense considering how fast the compiler is. Coming from C# I was assuming system libs were precompiled.

**Comment by JalonSolov** [comment]
If you only had the V compiler without the source... nothing much would work.  You wouldn't be able to compile anything without the `vlib` source tree.
